integration/docker/cpu_test.go

```diff
@@ -79,16 +85,18 @@ var _ = Describe("Hot plug CPUs", func() {
 
 	DescribeTable("container with CPU period and quota",
 		func(quota, period int, fail bool) {
+			vCPUs = ((quota + period - 1) / period) + defaultVCPUs
 			args = append(args, "--cpu-quota", fmt.Sprintf("%d", quota),
-				"--cpu-period", fmt.Sprintf("%d", period), Image, "sh", "-c", "sleep 5; nproc")
-			vCPUs = (quota + period - 1) / period
+				"--cpu-period", fmt.Sprintf("%d", period), Image, "sh", "-c",
+				fmt.Sprintf(`c=0; while [[ "$(cat %s/cpu%d/online 2> /dev/null)" != "1" ]] && [[ $c < %d ]]; do sleep %d; ((c++)); done; nproc --all`,
```

Comment:
I'd try (not sure if works) to move this to a mutliline string to make it more easy to read.